components/atom/tooltip/src/index.js

```diff
@@ -0,0 +1,234 @@
+import React, {Component, Fragment} from 'react'
+import PropTypes from 'prop-types'
+import {Tooltip} from 'reactstrap'
+import withIntersectionObserver from './hoc/withIntersectionObserver'
+
+const BASE_CLASS = 'sui-AtomTooltip'
+const CLASS_INNER = `${BASE_CLASS}-inner`
+const PREFIX_PLACEMENT = `${BASE_CLASS}-`
+const CLASS_TARGET = `${BASE_CLASS}-target`
+
+const PLACEMENTS = {
+  TOP: 'top',
+  TOP_START: 'top-start',
+  TOP_END: 'top-end',
+  RIGHT: 'right',
+  RIGHT_START: 'right-start',
+  RIGHT_END: 'right-end',
+  BOTTOM: 'bottom',
+  BOTTOM_START: 'bottom-start',
+  BOTTOM_END: 'bottom-end',
+  LEFT: 'left',
+  LEFT_START: 'left-start',
+  LEFT_END: 'left-end'
+}
+
+class AtomTooltip extends Component {
+  state = {isOpen: false}
+  preventNonTouchEvents = false
+  hasTouchEnded = false
+  touchTimer = null
+  refTooltip = React.createRef()
+  refTarget = React.createRef()
+
+  static defaultProps = {
```

Comment:
Could you please add this at the end together with the prop types? I'm afraid this gets overseen as we always add the prop types outside the class itself

Why are you setting it to true? Isn't it set by the intersection observer?

Reply:
If this `isVisible` is `false` or `undefined`, it cancels any other behavior of the tooltip. I've detected some cases where this was provoking an undesired behavior of the tooltip (wasn't working when it should) so setting this initial value fixed this. I can do more thorough tests to validate the removal of this `defaultProps`